test/lint/lint-submodule.py

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+"""
+This script checks for git modules
+"""
+
+import subprocess
+
+def main():
+    submodule_check_args = ['git', 'submodule', 'status', '--recursive',]
+    try:
+        submodules_list = subprocess.check_output(submodule_check_args, stderr = subprocess.STDOUT)
+    except subprocess.CalledProcessError:
+        exit(1)
```

Comment:
I think python already returns an exit code of 1 when it raises an exception

Reply:
I do think it's neater not to have a Python exception in the output, especially if the subprocess already prints its own self-evident errors.

Reply:
This should never happen, and when it happens, the process won't print it's own errors, so the exception seems better?

Reply:
I'm pretty sure that `git` prints its own error when it fails. But ok, no strong opinion here.

Reply:
`check_output` will eat the output. With `stderr = subprocess.STDOUT` it will also eat the error.

Reply:
I hadn't noticed the `stderr = subprocess.STDOUT`. Pretty sure we don't need it. The original script doesn't redirect the stderr either.

Reply:
Thank you @laanwj , @KevinMusgrave and @MarcoFalke for your review and points which led me to dig further.

I second @MarcoFalke  .  
2 cases:
1. Without `stderr = subprocess.STDOUT`, the error is printed onto the console and is not piped to Python.
2. With `stderr = subprocess.STDOUT`, if we want to access the error message, we can access it using `e.output` in the exception block (where `e` is CalledProcessError object). Otherwise, error is not printed onto console.

The reason why I used try exception blocks is to avoid the error getting printed. Additionally, I don't see any cases where this command will return exit code = 1. Should I remove try exception block ?
Otherwise, I think the code is good to go for now.

Kindly let me know your views too.

Reply:
I agree for (1) and (2), but why wouldn't we want the error piped to the console? The original script does no redirection of stderr either. You could capture it separately if you need the error message, but you don't. We want it it on the console for the CI.